apis/duck/v1/kresource_type.go

```diff
@@ -28,7 +28,7 @@ import (
 
 // KRShaped is an interface for retrieving the duck elements of an arbitrary resource.
 type KRShaped interface {
-	metav1.ObjectMetaAccessor
+	metav1.Object
 
 	GetTypeMeta() *metav1.TypeMeta
```

Comment:
I'd inline the metav1.TypeMeta methods we care about which should let you drop having downstream resources from implementing `GetTypeMeta()`

```suggestion
	SetGroupVersionKind(gvk schema.GroupVersionKind)
	GroupVersionKind() schema.GroupVersionKind
```

alternatively these satisfy the following interface

```suggestion
	schema.ObjectKind
```